Python/bytecodes.c

```diff
@@ -2726,6 +2726,29 @@ dummy_func(
             JUMPBY(INLINE_CACHE_ENTRIES_LOAD_ATTR);
         }
 
+        // error: LOAD_ATTR has irregular stack effect
+        inst(LOAD_ATTR_METHOD_MANAGED_DICT) {
+            assert(cframe.use_tracing == 0);
+            PyObject *self = TOP();
+            PyTypeObject *self_cls = Py_TYPE(self);
+            _PyLoadMethodCache *cache = (_PyLoadMethodCache *)next_instr;
+            uint32_t type_version = read_u32(cache->type_version);
+            assert(type_version != 0);
+            DEOPT_IF(self_cls->tp_version_tag != type_version, LOAD_ATTR)
+            assert(self_cls->tp_flags & Py_TPFLAGS_MANAGED_DICT);
+            PyDictOrValues dorv = *_PyObject_DictOrValuesPointer(self);
+            DEOPT_IF(_PyDictOrValues_IsValues(dorv), LOAD_ATTR);
+            PyDictKeysObject *keys = ((PyHeapTypeObject *)self_cls)->ht_cached_keys;
```

Comment:
This is the wrong keys. You need to check the object's dictionary's keys.

Reply:
Yes I just realised (see my message to you on discord :()

Reply:
Ah wait, I forgot that managed values means the object has no instance dict 🤦 . My bad.